aspnetcore/tutorials/razor-pages/razor-pages-start/sample/RazorPagesMovie50/Pages/Movies/Index.cshtml.cs

```diff
@@ -0,0 +1,88 @@
+﻿#define  SearchGenreAndString   //SearchString
+#region snippet_newProps
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using RazorPagesMovie.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace RazorPagesMovie.Pages.Movies
+{
+
+    public class IndexModel : PageModel
+    {
+        private readonly RazorPagesMovie.Data.RazorPagesMovieContext _context;
+
+        public IndexModel(RazorPagesMovie.Data.RazorPagesMovieContext context)
+        {
+            _context = context;
+        }
+
+        public IList<Movie> Movie { get; set; }
+        [BindProperty(SupportsGet = true)]
+        public string SearchString { get; set; }
+        // Requires using Microsoft.AspNetCore.Mvc.Rendering;
+        public SelectList Genres { get; set; }
+        [BindProperty(SupportsGet = true)]
+        public string MovieGenre { get; set; }
+        #endregion
+
+#if SearchString
+        #region snippet_1stSearch
+        public async Task OnGetAsync()
+        {
+            var movies = from m in _context.Movie
+                         select m;
```

Comment:
You should be able to simplify this LINQ statement to the following:

```suggestion
            var movies = _context.Movie;
```